src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/actions/CustomizablePanAction.java

```diff
@@ -89,62 +90,82 @@ public void setEnabled(boolean enabled) {
+        active = false;
+        return State.REJECTED;
+    }
+
     @Override
-    public State mousePressed (Widget widget, WidgetMouseEvent event) {
+    public State mousePressed(Widget widget, WidgetMouseEvent event) {
         EditorTopComponent editor = EditorTopComponent.getActive();
         if (editor != null) {
             editor.requestActive();
         }
-        if (isLocked ())
-            return State.createLocked (widget, this);
-        if (enabled && (event.getModifiersEx() & modifiersExMask) == modifiersEx) {
-            scene = widget.getScene ();
-            scrollPane = findScrollPane (scene.getView ());
+        if (isLocked()) {
+            return State.createLocked(widget, this);
+        }
+        if (enabled && (event.getModifiersEx() == modifiersEx)) {
+            scene = widget.getScene();
+            scrollPane = findScrollPane(scene.getView());
             if (scrollPane != null) {
-                lastLocation = scene.convertSceneToView (widget.convertLocalToScene (event.getPoint ()));
-                SwingUtilities.convertPointToScreen (lastLocation, scene.getView ());
-                return State.createLocked (widget, this);
+                rectangle = scene.getView().getVisibleRect();
+                lastLocation = scene.convertSceneToView(event.getPoint());
+                SwingUtilities.convertPointToScreen(lastLocation, scene.getView());
+                scene.getView().setCursor(Cursor.getPredefinedCursor(Cursor.MOVE_CURSOR));
+                return State.createLocked(widget, this);
             }
         }
         return State.REJECTED;
     }
 
-    private JScrollPane findScrollPane (JComponent component) {
+    private JScrollPane findScrollPane(JComponent component) {
         for (;;) {
-            if (component == null)
+            if (component == null) {
                 return null;
-            if (component instanceof JScrollPane)
+            }
+            if (component instanceof JScrollPane) {
                 return ((JScrollPane) component);
-            Container parent = component.getParent ();
-            if (! (parent instanceof JComponent))
+            }
+            Container parent = component.getParent();
+            if (!(parent instanceof JComponent)) {
                 return null;
+            }
             component = (JComponent) parent;
         }
     }
 
     @Override
-    public State mouseReleased (Widget widget, WidgetMouseEvent event) {
-        boolean state = pan (widget, event.getPoint ());
-        if (state)
+    public State mouseReleased(Widget widget, WidgetMouseEvent event) {
+        scene.getView().setCursor(Cursor.getDefaultCursor());
+        boolean state = pan(widget, event.getPoint());
+        if (state) {
             scrollPane = null;
-        return state ? State.createLocked (widget, this) : State.REJECTED;
+        }
+        return state ? State.REJECTED : State.REJECTED;
```

Comment:
That does not look correct.

Reply:
Thanks for catching that! You are right, this is not correct. 

I updated the PR to make the locking more clear:

`CustomizablePanAction` extends `WidgetAction.LockedAdapter` which is used for long-term actions like panning/moving. All methods return the `locked` state if `isLocked()` is true and `rejected` otherwise. 
The `locked` state means the event is processed and the processing has to stopped immediately (no other action should processed it). 
The `rejected` state the event is not processed by the action and has to be processed by other actions too.

In our case `CustomizablePanAction` should be `locked` when the user presses the mouse inside of the panning area (the graph). And `CustomizablePanAction` should be `unlocked` when the mouse button is released again. Leaving the panning area (`mouseExited`) and entering it again (`mouseEntered`) should not change the locking state as long as the user does not release the mouse.